config/config.go

```diff
@@ -276,26 +332,18 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			return fmt.Errorf("invalid rule file path %q", rf)
 		}
 	}
+
+	for _, sf := range c.ScrapeConfigFiles {
+		if !patRulePath.MatchString(sf) {
```

Comment:
`var patRulePath  = regexp.MustCompile(`^[^*]*(\*[^/]*)?$`)` 

when  var `fs`  equal  `conf.d/*/*.yml`  (i wanna it include all the sub dir yml files) that can't pass the regexp check. 
is't direct possible used the `filepath.Glob` to check the path ?

error:
```
ts=2023-03-07T03:48:08.128Z caller=main.go:468 level=error msg="Error loading config (--config.file=/var/tmp/prometheus/prometheus.yml)" file=/var/tmp/prometheus/prometheus.yml err="parsing YAML file /var/tmp/prometheus/prometheus.yml: invalid scrape config file path \"conf.d/*/*.yml\""
```

Reply:
This is how Prometheus does file globbing. It enables us to use inotify for file_sd and keeps all file globbing the same.